Good point.  I think I saw following, not strictly related problem (in stdlib), and looking at the Makefile I wrongly through it was caused by the d/i rules:

````
$ \rm -f camlheader camlheader_ur && make all -j 8
cl -nologo -D_CRT_SECURE_NO_DEPRECATE -O2 -Gy- -MD -DCAML_NAME_SPACE -c -I../byterun \
          -DRUNTIME_NAME='"ocamlrun"' headernt.c
cl -nologo -D_CRT_SECURE_NO_DEPRECATE -O2 -Gy- -MD -DCAML_NAME_SPACE -c -I../byterun \
          -DRUNTIME_NAME='"ocamlrun"' headernt.c
headernt.c
headernt.c
C:\cygwin\home\frisch\branches\v4\mlfi\ocaml\stdlib\headernt.c : fatal error C1083: Cannot open compiler generated file: 'C:\cygwin\home\frisch\branches\v4\mlfi\ocaml\stdlib\headernt.obj': Permission denied
make: *** [Makefile:162: camlheader_ur] Error 1
make: *** Waiting for unfinished jobs....
flexlink -merge-manifest -stack 16777216 -exe -o tmpheader.exe headernt.obj
rm -f camlheader.exe
mv tmpheader.exe camlheader
cp camlheader target_camlheader
cp camlheader camlheader_ur
````

I believe this is because make runs the same rule twice to build camlheader and camlheader_ur.  This can be fixed by adding copying rules for camlheader_ur (and target_camlheader), instead of copying them in the main rule.

I'll add that to this PR.

--- a/stdlib/Makefile
+++ b/stdlib/Makefile
@@ -160,20 +160,26 @@ else # Windows
 # TODO: see whether there is a way to further merge the rules below
 # with those above
 
-camlheader target_camlheader camlheader_ur: headernt.c
+camlheader: headernt.c
 	$(CC) -c $(CFLAGS) $(CPPFLAGS) -I../byterun \
 	          -DRUNTIME_NAME='"ocamlrun"' $(OUTPUTOBJ)headernt.$(O) $<
 	$(MKEXE) -o tmpheader.exe headernt.$(O) $(EXTRALIBS)
 	rm -f camlheader.exe
 	mv tmpheader.exe camlheader
+
+target_camlheader: camlheader
 	cp camlheader target_camlheader
+
+camlheader_ur: camlheader
 	cp camlheader camlheader_ur
 
-camlheaderd target_camlheaderd: headernt.c
+camlheaderd: headernt.c
 	$(CC) -c $(CFLAGS) $(CPPFLAGS) -I../byterun \
 	          -DRUNTIME_NAME='"ocamlrund"' $(OUTPUTOBJ)headerntd.$(O) $<
 	$(MKEXE) -o tmpheaderd.exe headerntd.$(O) $(EXTRALIBS)
 	mv tmpheaderd.exe camlheaderd
+
+target_camlheaderd: camlheaderd
 	cp camlheaderd target_camlheaderd
 
 camlheaderi: headernt.c

